### PR TITLE
Added support for DualShock 4 controller

### PR DESCRIPTION
**Pull request by ludokx**

### CODE REVIEW

File: backend/src/api/playstation.rs [hunk 1]
```diff
@@ -129,7 +249,7 @@ fn get_battery_status(charging_status: u8, battery_data: u8) -> BatteryInfo {
 
 #[cfg(test)]
 mod tests {
-    use crate::api::playstation::{DualSenseInputReport, DS_INPUT_REPORT_USB_SIZE};
+    use crate::api::playstation::{DualSenseInputReport, DualShock4InputReportCommon, Dualshock4InputReportUSB, Dualshock4InputReportBT, DS_INPUT_REPORT_USB_SIZE, DS4_INPUT_REPORT_USB_SIZE, DS4_INPUT_REPORT_BT_SIZE};
```
Comment (jfernandez) [inline backend/src/api/playstation.rs:252]: The rust-analyzer should complain that these imports are not being used right? Let's remove them until we add a test.

File: backend/src/api/playstation.rs [hunk 2]
```diff
@@ -57,13 +71,119 @@ struct DualSenseInputReport {
...
+    if cable_state > 0 {
+        if battery_data <= 10 {
+            charging_status = 0x1;
+        } else if battery_data == DS4_BATTERY_STATUS_FULL {
+            charging_status = 0x2;
+        } else {
+            charging_status = 0xf;
+        }
+    }
+    let battery_status = get_battery_status(charging_status, battery_data);
```
Comment (jfernandez) [inline backend/src/api/playstation.rs:181]: Neat that PS4 and PS5 controllers use the same logic to send battery status, but surprising that the Linux kernel doesn't re-use it.

**Review by jfernandez — APPROVED** [review]

File: backend/src/api/playstation.rs (hunk 1 above)
Comment (ludokx) [reply backend/src/api/playstation.rs]: yup I forgot to remove them, it just showed warnings and not errors but I removed them now

File: backend/src/api/playstation.rs (hunk 2 above)
Comment (ludokx) [reply backend/src/api/playstation.rs:181]: the logic is slightly different but in the end can be reused, DS has one slightly different state over DS4 but they still use the same values overall

**Pull request merged**